bot/src/abilities.ts

```diff
@@ -18,6 +22,31 @@ export const getDefaultOrderState = (): OrderState => {
   }
 }
 
+const updateProducts =
+  (prevProducts: Product[], productName: string, productQuantity: number): Product[] => {
+    // find existing product, if present
+    const existingProduct = prevProducts.find(x => x.name === productName)
+
+    // remove existing from array
+    const remainingProducts = prevProducts.filter((x: Product) => x.name !== productName)
+
+    let retProducts
+
+    // check if any Product with productName will still be in cart
+    if (productQuantity > 0 || (existingProduct && productQuantity + existingProduct.quantity > 0)) {
+      retProducts = [
+        ...remainingProducts,
+        { name: productName, quantity: (existingProduct ? existingProduct.quantity + productQuantity : productQuantity) }
```

Comment:
This will not update the product in place. The newly "updated" product will be added onto the array at the end of the array. Just making sure this is the desired result.